### ISSUE TITLE
Adding the code to SH2E

### ISSUE THREAD

**Issue by elishacloud** [Gemini-Loboto3/SH2config#1]
Polymega mentioned that you wanted details on how to integrate this code into the [Silent-Hill-2-Enhancements](https://github.com/elishacloud/Silent-Hill-2-Enhancements) main branch.  Below are the steps to do this:

1. Go to the [Silent-Hill-2-Enhancements](https://github.com/elishacloud/Silent-Hill-2-Enhancements) repository and click the "Fork" button in the upper right hand of the repository to fork the branch to your account.
2. Make changes to this forked Silent-Hill-2-Enhancements branch.
3. Once all the changes are made and it is ready to be merged back in then create a [pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request).

Let me know if you have any questions here.

**Comment by Gemini-Loboto3** [comment]
Ah, you meant as part of the DLL. Think I'll pass on that, it's easier to maintain it as its own thing.

**Comment by Polymega** [comment]
Elisha can you confirm if the launcher will essentially stay as its own standalone thing in the SH2E module? As in, when a user downloads SH2E it will now include the launcher exe as a new file in the download?

So the benefits of adding the code over to SH2E is to both include the launcher as part of the project's download but also try and use the project's code to help dynamically fill in XML data so we only have to maintain one master list of features?

**Comment by elishacloud** [comment]
> Ah, you meant as part of the DLL.

No.  We don't need this as part of the DLL.  However, I would like this launcher to be in the same GitHub repository.  I suppose I could just add it as a GitHub submodule so that it can remain in a separate repository and just linked to the main SH2E repository.

But actually having it part of the main repository makes it easier to share code (like the launch as admin code, the list of hidden options, etc).  This way as we change code, such as adding new hidden options or fix issues with shared code, we don't need to make sure to copy and paste the code into the launcher.

Also, having it integrated into the main SH2E repository makes it easier to add the launcher project file into the SH2E main Visual Studio solution which will allow us to build both the DLL and the launcher at the same time.

> Elisha can you confirm if the launcher will essentially stay as its own standalone thing in the SH2E module? As in, when a user downloads SH2E it will now include the launcher exe as a new file in the download?

Yes.  Having the launcher code integrated into the same repository does not mean it will be integrated into the same DLL.  We actually already have several other projects and smaller items in the repository, such as the [Silent Hill 2 Audio Enhancement Pack Builder](https://github.com/elishacloud/Silent-Hill-2-Enhancements/tree/master/AudioScripts).

**Comment by Gemini-Loboto3** [comment]
I'm okay with no integration. Works better in general, doesn't bloat the DLL, and pretty much mimics how it was also done on Silent Hill 3.

Problem with shared code is that the only two sources I used from EE are the [ini loader](https://github.com/Gemini-Loboto3/SH2config/blob/main/src/Settings.cpp) and the admin code (which I realized it's not committed yet), but I had to make changes for integration and trimmed anything that was unnecessary. The admin code probably won't need to be modified any further unless I bugged it, while the configuration stuff is all in the XML and is pretty much a separate thing. I designed it to be flexible enough for future expansions, so all it requires are minor adjustments to the XML to support any new options and that's it.

**Comment by elishacloud** [comment]
> I'm okay with no integration. Works better in general, doesn't bloat the DLL, and pretty much mimics how it was also done on Silent Hill 3.

Yeah, I don't want to add the code into the DLL.  However, it would be nice if the VS project file was integrated into the SH2E VS solution such that when I compile the SH2E build it creates both the DLL and the launcher at the same time.  The code, of course, would still be separate.

> the only two sources I used from EE are the [ini loader](https://github.com/Gemini-Loboto3/SH2config/blob/main/src/Settings.cpp) and the admin code

I actually have some code in the [AutoUpdate](https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/742de6e59d64be15a63b9ab6e4613d8a4cabbb5a/Common/AutoUpdate.cpp#L267-L328) that might be useful for the ini loader/updater.  The code here would allow the ini file to maintain the default remarks and structure.  Anyways, I can work on that later, but shows another example of what could be done if the code was part of the main repository.

BTW: if you are worried about merging it into the main SH2E repository then I can do it, no problem, once you are done with it.  You don't need to worry about that if you don't want to.  I just don't want to take your code and merge it in without your approval.  :-)
